src/app/services/actions/query-action-creators.ts

```diff
@@ -31,9 +32,20 @@ export function runQuery(query: IQuery): Function {
       }
     }
 
-    const headers = { Authorization: `Bearer ${authToken}` };
+    const headers = {
+      'Authorization': `Bearer ${authToken}`,
+      'Content-Type': 'application/json',
+    };
 
-    return fetch(graphUrl, { headers })
+    let options: IRequestOptions = {};
+    options = { method: query.selectedVerb, headers};
+
+    if (query.sampleBody && query.selectedVerb !== 'GET') {
+      const body = JSON.stringify(query.sampleBody);
+      options.body = body;
```

Comment:
Mutating objects like this can be a source of hard to debug bugs. You can instead create a new object and assign it to options.

options = Object.assign({}, options, { body: body })